backend/tests/controllers/artefacts/test_artefacts_search.py

```diff
@@ -0,0 +1,220 @@
+    other_artefact = generator.gen_artefact(name=f"other_artefact_{unique_marker}")
+
+    # Search for the specific pattern
+    response = test_client.get("/v1/artefacts/search?q=special_search")
+
+    assert response.status_code == 200
+    data = response.json()
+    artefacts = data["artefacts"]
+
+    # Should only find the target artefact
+    assert target_artefact.name in artefacts
+    assert other_artefact.name not in artefacts
+
+
+def test_search_artefacts_case_insensitive(
+    test_client: TestClient, generator: DataGenerator
+):
+    """Test that search is case-insensitive"""
+    unique_marker = uuid.uuid4().hex[:8]
+    artefact_name = f"MixedCase_Artefact_{unique_marker}"
+
+    generator.gen_artefact(name=artefact_name)
+
+    # Search with different cases
+    for search_term in ["mixedcase", "MIXEDCASE", "MixedCase"]:
+        response = test_client.get(f"/v1/artefacts/search?q={search_term}")
+        assert response.status_code == 200
+        artefacts = response.json()["artefacts"]
+        assert artefact_name in artefacts
+
+
+def test_search_artefacts_partial_match(
+    test_client: TestClient, generator: DataGenerator
+):
+    """Test that search supports partial matching"""
+    unique_marker = uuid.uuid4().hex[:8]
+    artefact_name = f"ubuntu_desktop_{unique_marker}"
+
+    generator.gen_artefact(name=artefact_name)
+
+    # Search with partial term
+    response = test_client.get("/v1/artefacts/search?q=desktop")
+
+    assert response.status_code == 200
+    artefacts = response.json()["artefacts"]
+    assert artefact_name in artefacts
+
+
+def test_search_artefacts_pagination(test_client: TestClient, generator: DataGenerator):
+    """Test that pagination works correctly"""
+    unique_marker = uuid.uuid4().hex[:8]
+    names = []
+
+    # Create 10 artefacts
+    for i in range(10):
+        name = f"paginated_artefact_{unique_marker}_{i:02d}"
+        generator.gen_artefact(name=name)
+        names.append(name)
+
+    expected_sorted = sorted(names)
+
+    # Get first page
+    response = test_client.get(
+        f"/v1/artefacts/search?q=paginated_artefact_{unique_marker}&limit=3&offset=0"
+    )
+    assert response.status_code == 200
+    page1 = response.json()["artefacts"]
+    assert len(page1) == 3
+    assert page1 == expected_sorted[0:3]
+
+    # Get second page
+    response = test_client.get(
+        f"/v1/artefacts/search?q=paginated_artefact_{unique_marker}&limit=3&offset=3"
+    )
+    assert response.status_code == 200
+    page2 = response.json()["artefacts"]
+    assert len(page2) == 3
+    assert page2 == expected_sorted[3:6]
+
+
+def test_search_artefacts_limit_validation(test_client: TestClient):
+    """Test that limit parameter is validated"""
+    # Test maximum limit
+    response = test_client.get("/v1/artefacts/search?limit=1001")
+    assert response.status_code == 422
+
+    # Test minimum limit
+    response = test_client.get("/v1/artefacts/search?limit=0")
```

Comment:
Usually the `limit=0` means that no pagination is applied and all the results are returned in one request. is this behaviour supported?